python/ray/tune/tests/execution/test_controller_callback_integration.py

```diff
@@ -52,7 +52,7 @@ def test_callback_save_restore(
         runner._callbacks.on_trial_result(
             iteration=i, trials=None, trial=None, result=None
         )
-    runner.checkpoint(force=True)
+    runner.checkpoint(force=True, wait=True)
```

Comment:
Is there ever a case we want to support `wait=False` anymore?

Reply:
`wait=False` is what happens for forceful `num_to_keep` based syncing. However, these unit tests rely on restoring the tune controller state right after, so we should actually wait for the sync to finish.